Update/email editor namespaces

lysyjan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #55602.

- This PR replaces the MailPoet word in constants and names for WooCommerce. 
- Replace prefixes in class names from `mailpoet` to `woocommerce`.
- Removes unused type and stubs related to a usage Codeception in tests.
- Unify file and class names for the files classes added recently.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block_email_editor experimental flag (in DB, set woocommerce_feature_block_email_editor_enabled option to yes).
2. Navigate to WP Admin > Woo Emails
3. Verify that email editor works as expected after renaming

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

Those changes do not need a changelog entry because it does not change any funcionality.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

lysyjan committed:
Refactor: Update email editor namespace to WooCommerce

This commit updates the namespace for the email editor package from MailPoet\EmailEditor to Automattic\WooCommerce\EmailEditor across all files. The changes include:

lysyjan committed:
Refactor  action and filters prefix from mailpoet to woocommerce

lysyjan committed:
Refactor JS action and filters prefix from mailpoet to woocommerce

#55602

lysyjan committed:
Rename filters namespace prefixes from mailpoet to woocommerce

#55602

lysyjan committed:
Change ID element for initilize email editor element

#55602

lysyjan committed:
Change mailpoet in ID names and constants to woocommerce

#55602

lysyjan committed:
Change comments mentioning MailPoet package

Comments mentioned MailPoet which is not anymore related after moving into Woo repository.
This commit replaces MailPoet with WooCommerce.
#55602

lysyjan committed:
Rename base class for email editor unit tests

#55602

lysyjan committed:
Replace mailpoet CSS class names with woocommerce

This commit updates CSS class names across multiple files in the email editor package, replacing 'mailpoet' prefixes with 'woocommerce'. The changes include:

- Updating class names in SCSS and TSX files
- Ensuring consistent naming convention across components
- Maintaining the existing styling and functionality

These changes are part of the ongoing effort to migrate the email editor package from MailPoet to WooCommerce.

#55602

lysyjan committed:
Update README.md for WooCommerce Email Editor package

This commit updates the README.md file to reflect the transition from MailPoet to WooCommerce Email Editor package. Changes include:

- Updating package name and description
- Modifying composer script commands
- Updating file paths and references
- Adjusting comments about package-specific integrations
- Updating filter documentation to reference WooCommerce instead of MailPoet

#55602

lysyjan committed:
Rename MailPoet global object to WooCommerce in email editor

This commit updates the global window object from `MailPoetEmailEditor` to `WooCommerceEmailEditor` across multiple files in the email editor package. Changes include:

- Updating global object references in TypeScript files
- Modifying PHP page renderer to use the new global object name
- Ensuring consistent naming across the email editor implementation

#55602

lysyjan committed:
Update namespace references from MailPoet to WooCommerce in email editor files

This commit updates the namespace references in several PHP files within the email editor package, replacing MailPoet namespaces with WooCommerce namespaces. Changes include:

- Updating import statements in Integration.php, Package.php, PageRenderer.php, and other related files
- Modifying filter hook names to use 'woocommerce_' prefix instead of 'mailpoet_'
- Ensuring consistent namespace usage across the email editor implementation

#55602

lysyjan committed:
Remove IntegrationTester and UnitTester support files

This commit removes the IntegrationTester.php and UnitTester.php files from the email editor package's test support directory. These files appear to be no longer needed as part of the ongoing migration from MailPoet to WooCommerce Email Editor.

#55602

lysyjan committed:
Update translation domain from MailPoet to WooCommerce in email editor files

This commit updates the translation domain from 'mailpoet' to 'woocommerce' across multiple PHP files in the email editor package. Changes include:

- Updating __() and esc_html__() translation function domains
- Ensuring consistent translation domain usage
- Preparing for WooCommerce-specific localization

#55602

lysyjan committed:
Update translation domain from MailPoet to WooCommerce in email editor components

This commit updates the translation domain from 'mailpoet' to 'woocommerce' across multiple TypeScript and TSX files in the email editor package. Changes include:

- Updating __() translation function domains in components
- Ensuring consistent translation domain usage
- Preparing for WooCommerce-specific localization

#55602

lysyjan committed:
Update keyboard shortcut names from MailPoet to WooCommerce

Replace 'mailpoet/email-editor' prefix with 'woocommerce/email-editor' in keyboard shortcut names and related function calls. This change ensures consistency with the ongoing migration of the email editor package from MailPoet to WooCommerce.

#55602

lysyjan committed:
Remove MailPoet-specific type definition from email editor types

Remove the `MailPoetEmailPostContentExtended` type from the email editor types file as part of the ongoing migration from MailPoet to WooCommerce Email Editor.

#55602

lysyjan committed:
Unify email editor package file naming

This commit introduces several structural changes to the email editor package:

- Renamed Bootstrap.php to class-bootstrap.php with improved code style
- Added new class-email-editor-container.php for dependency injection
- Replaced email-css-inliner.php with class-email-css-inliner.php
- Added class-package.php to manage package initialization and metadata
- Updated class naming to follow WordPress coding standards (e.g., Email_Css_Inliner)
- Improved code documentation and type hinting

These changes improve code organization and maintainability while continuing the migration from MailPoet to WooCommerce Email Editor.

#55602

triple0t commented on packages/php/email-editor/src/Engine/class-email-editor.php:
```diff
@@ -21,7 +21,7 @@
  */
 class Email_Editor {
-	public const MAILPOET_EMAIL_META_THEME_TYPE = 'mailpoet_email_theme';
+	public const WOOCOMMERCE_EMAIL_META_THEME_TYPE = 'woocommerce_email_theme';
```
I believe this change will prevent previously saved global styles from working. 
This will affect the following clients
* MailPoet
* Growth infra
* Email editor users on MailPoet.


Can we find a way to migrate this data to them or a less stressful way to pass this information on?

lysyjan replied:
In the MailPoet plugin, we can prepare a migration. That's not a big deal, I guess.
I am not sure about Growth infra. What is their usage now?

triple0t replied:
I don't know. But we can check with Patrick.
Please indicate this as a breaking change on the email editor package changelog

triple0t commented on packages/php/email-editor/src/Engine/Templates/class-templates.php:
```diff
@@ -20,7 +20,7 @@ class Templates {
 	 * @var string $plugin_slug
 	 */
-	private string $template_prefix = 'mailpoet';
+	private string $template_prefix = 'woocommerce';
```
Can we also remove the [powered-by-mailpoet ](https://github.com/woocommerce/woocommerce/blob/1659737b73147adae6a8a3a69ede4b7e96b59f3f/packages/php/email-editor/src/Engine/Templates/email-general.html#L2) from the default template?

triple0t commented on packages/php/email-editor/src/Engine/class-theme-controller.php:
```diff
@@ -222,14 +222,14 @@ public function get_stylesheet_for_rendering( ?WP_Post $post = null, $template =
 		// Because the section styles is not a part of the output the `get_styles_block_nodes` method, we need to get it separately.
 		if ( $template && $template->wp_id ) {
-			$template_theme    = (array) get_post_meta( $template->wp_id, 'mailpoet_email_theme', true );
+			$template_theme    = (array) get_post_meta( $template->wp_id, 'woocommerce_email_theme', true );
```
Please check my [former comment](https://github.com/woocommerce/woocommerce/pull/55938#discussion_r1975506436) on this.
Also, can we reuse the existing constant (`WOOCOMMERCE_EMAIL_META_THEME_TYPE`) here

triple0t commented on packages/php/email-editor/src/Engine/class-user-theme.php:
```diff
@@ -15,7 +15,7 @@
  */
 class User_Theme {
-	private const USER_THEME_POST_NAME = 'wp-global-styles-mailpoet-email';
+	private const USER_THEME_POST_NAME = 'wp-global-styles-woocommerce-email';
```
The [same note](https://github.com/woocommerce/woocommerce/pull/55938#discussion_r1975506436) here

triple0t commented on packages/php/email-editor/tests/_support/IntegrationTester.php:
```diff
@@ -80,7 +80,7 @@ public function cleanup(): void {
 	 */
 	public function cleanup_user_theme_post(): void {
-		$post = get_page_by_path( 'wp-global-styles-mailpoet-email', OBJECT, 'wp_global_styles' );
+		$post = get_page_by_path( 'wp-global-styles-woocommerce-email', OBJECT, 'wp_global_styles' );
```
Let's reuse the constant `USER_THEME_POST_NAME` for `'wp-global-styles-woocommerce-email'`

lysyjan replied:
This file was removed in this commit: 98cd655bc3deeba7bae67ddf35b2600933edde56

triple0t commented on plugins/woocommerce/src/Internal/EmailEditor/Package.php:
```diff
@@ -42,7 +42,7 @@ final public static function init() {
 		}
 
-		\MailPoet\EmailEditor\Package::init(); // The namespace will be changed to Automattic\WooCommerce\EmailEditor\Package in the future.
+		\Automattic\WooCommerce\EmailEditor\Package::init(); // The namespace will be changed to Automattic\WooCommerce\EmailEditor\Package in the future.
```
You can remove the comment here

triple0t reviewed (commented):
Well done. @lysyjan  👏🏾
This is incredible and nice to see.

I have a major blocker with this PR; the rest are minor notes.

With the global style name update, any previously saved global style will stop working. This would have been fine before, but unfortunately, we now have clients using the Email Editor, so we need to consider them when implementing this change.

Can we think of a migration path or implement something that will gracefully move the old global styles to the new styles key?

Other minor notes
1. Can we remove the `powered-by-mailpoet` block on the default template
2. Fix the Save label title on the Woo core side integration (here: https://github.com/woocommerce/woocommerce/blob/711f6845ffdf0131d5a1235d09b0108f6974ed8b/plugins/woocommerce/client/admin/client/wp-admin-scripts/email-editor-integration/index.ts#L11)
3. We missed this: https://github.com/woocommerce/woocommerce/blob/c07905b0e54465388bc50da8bfd03bf0902a378a/packages/php/email-editor/src/Engine/Templates/class-templates-registry.php#L62
4. We missed these on Integration test files: 
https://github.com/woocommerce/woocommerce/blob/c07905b0e54465388bc50da8bfd03bf0902a378a/packages/php/email-editor/tests/integration/Engine/Templates/Templates_Test.php, 
https://github.com/woocommerce/woocommerce/blob/c07905b0e54465388bc50da8bfd03bf0902a378a/packages/php/email-editor/tests/unit/Engine/Templates/Template_Registry_Test.php
5. WP PHP Codestyle errors: https://gist.github.com/triple0t/2390116e3ec1aa84d296d04c4afde5b2 
You can test by replacing 
```
		"code-style": "../../../mailpoet/tasks/code_sniffer/vendor/bin/phpcs -ps",
		"code-style-fix": "../../../mailpoet/tasks/code_sniffer/vendor/bin/phpcbf -p",
```
with
```
		"code-style": "../../../plugins/woocommerce/bin/composer/phpcs/vendor/squizlabs/php_codesniffer/bin/phpcs -ps",
		"code-style-fix": "../../../plugins/woocommerce/bin/composer/phpcs/vendor/squizlabs/php_codesniffer/bin/phpcbf -p",
```

lysyjan committed:
Remove redundant comment in Email Editor Package initialization

Clean up the Package.php file by removing an unnecessary comment about future namespace changes. This simplifies the code and removes outdated documentation.

#55602

lysyjan committed:
Remove MailPoet Powered By block from email template

Remove the MailPoet-specific "Powered By" block from the general email template, continuing the migration from MailPoet to WooCommerce Email Editor.

#55602

lysyjan committed:
Update test files to use WooCommerce email template references

Replace MailPoet-specific template references with WooCommerce email template names in integration and unit tests. This continues the migration from MailPoet to WooCommerce Email Editor by updating template initialization and naming in test cases.

#55602

lysyjan committed:
Update email editor integration filter hook name

Replace 'mailpoet_email_editor_send_button_label' with 'woocommerce_email_editor_send_button_label' in the email editor integration script. This continues the migration from MailPoet to WooCommerce Email Editor by updating filter hook names.

#55602

lysyjan committed:
Update theme meta key for email editor template retrieval

Replace hardcoded meta key with constant from Email_Editor class when retrieving template theme metadata. This improves code maintainability and consistency in the theme controller.

#55602

lysyjan committed:
Update template example in Templates Registry documentation

Replace MailPoet-specific template name with WooCommerce email template name in the get_by_name() method documentation. This continues the migration from MailPoet to WooCommerce Email Editor by updating code comments.

#55602

lysyjan committed:
Update code style scripts to use WooCommerce PHPCS paths

Replace MailPoet-specific code style script paths with WooCommerce PHPCS vendor paths in composer.json. This continues the migration from MailPoet to WooCommerce Email Editor by updating development tooling references.

#55602

lysyjan committed:
Improve code formatting

This commit includes several minor code style improvements:
- Adjust whitespace and indentation in various files
- Add phpcs disable comment in bootstrap file
- Minor formatting consistency updates in test files

These changes enhance code readability and maintain consistent formatting across the email editor package.

#55602

lysyjan committed:
Update email theme post type name

Rename the email theme post type from `mailpoet_email_theme` to `woocommerce_email_theme` to align with WooCommerce naming conventions.

#55602

lysyjan committed:
Update user theme post name for WooCommerce email editor

Rename the user theme post name from `wp-global-styles-mailpoet` to `wp-global-styles-woocommerce-email` to align with WooCommerce naming standards.

#55602

lysyjan commented:
@triple0t I hope I addressed all your remarks. Thank you for the detailed review.

I thought the point about the code sniffer should be a part of the another issue: https://github.com/woocommerce/woocommerce/issues/55788
Nevertheless, I address it here.

triple0t reviewed (approved):
I tested it and can confirm it works well.

![Screenshot 2025-03-04 at 1 06 18 PM](https://github.com/user-attachments/assets/b8e95f8d-ed30-4663-89a4-8588c50f4865)
The editor loads.

![Screenshot 2025-03-04 at 1 16 50 PM](https://github.com/user-attachments/assets/541afe47-1da8-4931-9be0-81c2ed62026c)
No new error in the browser console.

![Screenshot 2025-03-04 at 1 06 41 PM](https://github.com/user-attachments/assets/ede9c67f-f796-4f34-8e69-ca4fdea4ce5c)
The translation command no longer shows an error.

And the Namespace was successfully updated to `Automattic\WooCommerce\EmailEditor`

Well done.
